packages/create-react-app/index.js

```diff
@@ -108,38 +108,47 @@ function createApp(name, verbose, version) {
 }
 
 function install(packageToInstall, verbose, callback) {
-  var args = [
-    'add',
-    '--dev',
-    '--exact',
-    packageToInstall,
-  ];
-  var proc = spawn('yarn', args, {stdio: 'inherit'});
-
-  var yarnExists = true;
-  proc.on('error', function (err) {
-    if (err.code === 'ENOENT') {
-      yarnExists = false;
-    }
-  });
-  proc.on('close', function (code) {
-    if (yarnExists) {
-      callback(code, 'yarn', args);
-      return;
-    }
-    // No Yarn installed, continuing with npm.
-    args = [
+  function fallbackToNpm() {
+    var npmArgs = [
       'install',
       verbose && '--verbose',
       '--save-dev',
       '--save-exact',
       packageToInstall,
     ].filter(function(e) { return e; });
-    var npmProc = spawn('npm', args, {stdio: 'inherit'});
+    var npmProc = spawn('npm', npmArgs, {stdio: 'inherit'});
     npmProc.on('close', function (code) {
-      callback(code, 'npm', args);
+      callback(code, 'npm', npmArgs);
     });
-  });
+  }
+
+  var yarnArgs = [
+    'add',
+    '--dev',
+    '--exact',
+    packageToInstall,
+  ];
+  var yarnProc;
+  var yarnExists = true;
+  try {
+    yarnProc = spawn('yarn', yarnArgs, {stdio: 'inherit'});
```

Comment:
I'd only wrap this single line with `try-catch`, because that's the thing we need to catch the errors from. I don't want us to get stuck in an infinite loop if something else fails in the fallback `npm` command for example.